lib/components/Thumbs.js

```diff
@@ -48,14 +48,18 @@ module.exports = React.createClass({
         window.removeEventListener("DOMContentLoaded", this.updateStatics);
     },
     componentDidMount: function componentDidMount(nextProps) {
+        var that = this;
         // as the widths are calculated, we need to resize
         // the carousel when the window is resized
         window.addEventListener("resize", this.updateStatics);
         // issue #2 - image loading smaller
         window.addEventListener("DOMContentLoaded", this.updateStatics);
 
         var defaultImg = ReactDOM.findDOMNode(this.thumb0).getElementsByTagName('img')[0];
-        defaultImg.addEventListener('load', this.setMountState);
+        defaultImg.addEventListener('load', function () {
+            that.setMountState();
+            that.updateStatics();
```

Comment:
Just a nitpick, can you rename `setMountState` to `onFirstImageLoad`, and call `updateStatics` inside that method?